common/test_filestore_utils.py

```diff
@@ -18,30 +18,110 @@
 from common import filestore_utils
 from common import new_process
 
-# TODO(zhichengcai): Figure out how we can test filestore_utils when using
-# the local_filestore implementation.
+LOCAL_DIR = '/dir'
+LOCAL_DIR_2 = '/dir2'
+GCS_DIR = 'gs://fake_dir'
+GCS_DIR_2 = 'gs://fake_dir_2'
 
 
-def test_keyword_args():
+def test_using_local_filestore(fs, use_local_filestore):  # pylint: disable=invalid-name,unused-argument
+    """Tests that local_filestore is used in local running settings."""
+    fs.create_dir(LOCAL_DIR)
+    fs.create_dir(LOCAL_DIR_2)
+
+    with mock.patch('common.new_process.execute') as mocked_execute:
+        filestore_utils.cp(LOCAL_DIR, LOCAL_DIR_2, recursive=True)
+        assert 'gsutil' not in mocked_execute.call_args_list[0][0][0]
```

Comment:
Can you use pdb and figure out what `call_args_list[0][0]` is? If it's just the command than we don't need [0][0][0].

Reply:
It seems we do need [0][0][0], otherwise it is not the actually argument list: 
```
-> assert 'gsutil' not in mocked_execute.call_args_list[0][0][0]
(Pdb) p mocked_execute.call_args_list
[call(['cp', '-r', '/dir', '/dir2'], expect_zero=True)]
(Pdb) p mocked_execute.call_args_list[0]
call(['cp', '-r', '/dir', '/dir2'], expect_zero=True)
(Pdb) p mocked_execute.call_args_list[0][0]
(['cp', '-r', '/dir', '/dir2'],)
(Pdb) p mocked_execute.call_args_list[0][0][0]
['cp', '-r', '/dir', '/dir2']
(Pdb)
```

Reply:
ACK